src/components/ButtonGroupWithDropdown/ButtonGroupWithDropdown.tsx

```diff
@@ -0,0 +1,75 @@
+import { Box, Button, ChervonDownIcon, Dropdown, List, Text } from "@saleor/macaw-ui-next";
+import React from "react";
+
+interface ButtonGroupWithDropdownProps {
+  onClick?: () => void;
+  options: Array<{
+    label: string;
+    testId?: string;
+    onSelect: <T>(params?: T) => void;
+  }>;
+  testId?: string;
+  disabled?: boolean;
+}
+
+// TODO: consider moving this to Macaw UI
+export const ButtonGroupWithDropdown: React.FC<ButtonGroupWithDropdownProps> = ({
+  children,
+  options,
+  onClick,
+  disabled = false,
+  testId,
+}) => {
+  return (
+    <Dropdown>
+      <Box display="flex">
+        <Button
+          onClick={onClick}
+          data-test-id={testId}
+          disabled={disabled}
+          // TODO: fix this in Macaw UI - allow overriding border radius
+          style={{
+            borderBottomRightRadius: 0,
+            borderTopRightRadius: 0,
+          }}
+        >
+          {children}
+        </Button>
+
+        <Dropdown.Trigger>
+          <Button
+            icon={<ChervonDownIcon />}
+            disabled={disabled}
+            borderColor="default1"
```

Comment:
Maybe it should be transparent, so it's like the background?

Reply:
I've checked and it didn't work
![2025-06-03 at 11 24 41 - CleanShot@2x](https://github.com/user-attachments/assets/1e0a80c9-894b-4f1b-9140-ed72e0c5cd5d)
afaik `default1` is the background color we use in Dashboard anyway 😄